### PR TITLE
Adds a new detective and spy thief syndicate item: The scuttlebot!

### PR DESCRIPTION
**Pull request by Bartimeus973**
## About the PR 
Adds the fake detective hat, scuttlebot and remote control glasses to the game. Costs 2 TC.
The fake detective hat is a detective hat with a different description, that when used in hand turns into the glasses and the scuttlebot itself. Using the glasses on the scuttlebot turns it back into the hat.
Wearing the glasses puts you in control of the scuttlebot and lets you spy on things.

The scuttlebot is a weak critter with one hand slot and three abilities. It fits under tables and squeezes through doors.

scuttlebot

Take picture lets you print a photo like a camera.

hatpicture

Flash lets you (unsurpringly) flash someone at melee range

hatflash

Return to owner lets you return to your body

shutdown

## Why's this needed? 
More ways to gather information. Opens up a lot of room for gimmicks. Lets you go on a recon mission without risking your life. Can be used to take pictures of stuff as a spy and get more bounties. Fits the theme of detective and spy thief the most!

Special thanks to shiiba/bork5796 for their feedback on the sprites!

## Changelog

```changelog
(u)Bartimeus973
(*)The detective and spy thieves now have access to a new item: The scuttlebot! A strangely hat-shaped robot that is eager to spy on everything and everyone.
```

### CODE REVIEW

**Comment by zjdtmkhzt** [comment]
I love the thought of the detective trying to be all sneaky with this and leaving no trail back to them, meanwhile the thing is literally a detective hat with only one detective on station.
But luckily spy thieves can get it as well!

File: code/datums/abilities/critter/scuttlebot.dm [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+					mob_title = "[M]"
+				else
+					mob_title += " and [M]"
+
+				if (mobnumber < 4)
+					var/holding = null
+					if (iscarbon(M))
+						var/mob/living/carbon/temp = M
+						if (temp.l_hand || temp.r_hand)
+							var/they_are = M.gender == "male" ? "He's" : M.gender == "female" ? "She's" : "They're"
```
Comment (zjdtmkhzt) [inline code/datums/abilities/critter/scuttlebot.dm:49]: This is what we have hes_or_shes() for!
```

**Review by zjdtmkhzt — COMMENTED** [review]
I'm not saying *you* have to do it, but the photo-taking-code seems like something we should really unify in one proc and modernize? It doesn't even use the pronoun procs!

**Comment by Bartimeus973** [comment]
I'll admit to taking the photo code straight out of the camera object.

**Comment by Tarmunora** [comment]
> I love the thought of the detective trying to be all sneaky with this and leaving no trail back to them, meanwhile the thing is literally a detective hat with only one detective on station. But luckily spy thieves can get it as well!

this seems... amusingly useless for this reason yes. I know there have been concerns in the past about bloating syndicate item lists with stuff that is unlikely to really see use. 🤷‍♀️

**Comment by Tarmunora** [comment]
Maybe bump the limb's max w_class up to at least W_CLASS_SMALL - would open up the ability for it to hold a silenced .22 or the det revolver, which, while not really the spirit of the item, i think would be an interesting tactic that would be worth enabling

**Comment by Bartimeus973** [comment]
Done, the bot can now hold a lot more items including small lethal firearms for your antagonistic needs.

**Pull request merged**